pkg/controllers/inventory/inventory_controller.go

```diff
@@ -169,5 +170,39 @@ func (c *InventoryController) syncInventoryKeys() {
 
 func (c *InventoryController) CleanStaleInventoryObjects() error {
 	log.Info("Clean stale inventory objects")
+	err := c.CleanStaleInventoryApplicationInstance()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *InventoryController) CleanStaleInventoryApplicationInstance() error {
```

Comment:
Move it to services/inventory/pod.go, keep this file not specific CR related.

Reply:
Could you refactor it to 
CleanupInventoryApplicationInstance(onlyStale bool)
then we can reuse it in the following Cleanup feature.

if onlyStale, keep this logic as it is, if all, then not to check c.service.IsPodDeleted.

Reply:
Good idea. move it to pod.go
The cleanup/gc has different logic. it seems no need to merge them.